setup.py

```diff
@@ -21,7 +21,7 @@
     setup_requires=["setuptools_scm~=6.0"],
     install_requires=[
         "z-quantum-core",
-        "pyquil>=2.25.0",
+        "pyquil~=2.25.0",
```

Comment:
@Shiro-Raven I know this might be too late because the PR was already merged, but shouldn't we change this to `pyquil~=2.25`? Including the trailing patch will cause the constraint to reject newer minor releases like `2.26`